### PR TITLE
Keep URL path if falling back to the default site

### PR DESCRIPTION
**Pull request by dcsaszar**
This leads to better fallbacks for outdated/legacy permalinks, e.g.,

* https://example.com/contact -> https://example.com/en/contact (instead of https://example.com/en)
* https://example.com/en-US/contact -> https://example.com/en/contact (instead of https://example.com/en)

### CODE REVIEW

File: src/config/scrivitoSites.ts [hunk 1]
```diff
@@ -84,24 +85,40 @@ export function isNoSitePresent(): boolean {
...
+    return
+  }
+
+  const site = await load(getPreferredSite)
+  if (!site) return
+
+  const language = languageForUrl(window.location.href)
+  const slashesCount = language ? 2 : 1
+  const prefixLength = (language?.length || 0) + slashesCount
+  const path = window.location.pathname.substring(prefixLength)
```
Comment (apepper) [inline src/config/scrivitoSites.ts:101]: Why do we need this logic? It looks quite complex to me.

File: src/config/scrivitoSites.ts [hunk 2]
```diff
@@ -84,24 +85,40 @@ export function isNoSitePresent(): boolean {
...
+
+  if (!path) {
+    navigateTo(site)
+    return
   }
+
+  const url = await load(() => urlFor(site))
+  window.location.assign([url, path].join(url.endsWith('/') ? '' : '/'))
```
Comment (apepper) [inline src/config/scrivitoSites.ts:109]: If I'm not mistaken query parameter and hashes would not be transferred to the new URL. Is this intentional?

File: src/config/scrivitoSites.ts (hunk 1 above)
Comment (apepper) [reply src/config/scrivitoSites.ts:101]: If we don't need it the logic could swap `location.origin` from `location.href` with `urlFor(site)`.

**Review by apepper — COMMENTED** [review]
FEEDBACK ([c201e4c](https://github.com/Scrivito/scrivito-portal-app/pull/708/commits/c201e4c506f4b3bd0c627a3d027038887de33346))

Very nice 🚀.

Please see my comments to maybe simplify this PR.

File: src/config/scrivitoSites.ts (hunk 2 above)
Comment (dcsaszar) [reply src/config/scrivitoSites.ts:109]: Yes and no. I only cared about permalink-like URLs b/c the original feature request only mentioned these. Everything else works or doesn't work by chance.

File: src/config/scrivitoSites.ts (hunk 1 above)
Comment (dcsaszar) [reply src/config/scrivitoSites.ts:101]: Depends on which kinds of old permalinks we want to handle:
1. links after relaunching a single-language site
2. links to the primary language site version that had no language prefix before the relaunch
3. links to a language version that no longer exists after the relaunch

The current logic handles all cases. If we drop (3) the logic is simpler.

Comment (apepper) [reply src/config/scrivitoSites.ts:101]: I would drop (3) to keep it more simple.

File: src/config/scrivitoSites.ts [hunk 3]
```diff
@@ -84,24 +84,33 @@ export function isNoSitePresent(): boolean {
...
+  }
+
+  const site = await load(getPreferredSite)
+  if (!site) return
+
+  const siteUrl = await load(() => urlFor(site))
+  const { pathname, search, hash } = window.location
+  const path = pathname === '/' ? '' : pathname
+
+  window.location.assign(`${siteUrl}${path}${search}${hash}`)
```
Comment (apepper) [inline src/config/scrivitoSites.ts:101]: Will `siteUrl` always contain a trailing `/`? In your prior code you checked for it to end with `/` or not.

**Review by apepper — COMMENTED** [review]
FEEDBACK ([e9dbe0d](https://github.com/Scrivito/scrivito-portal-app/pull/708/commits/e9dbe0d7994eb0d91eb3b8752246566785d6caeb))

Really nice. I left one smaller comment.

File: src/config/scrivitoSites.ts (hunk 3 above)
Comment (dcsaszar) [reply src/config/scrivitoSites.ts:101]: This time it's the other way around: pathname will always **start** with a `/` (except if it's empty)

Comment (apepper) [reply src/config/scrivitoSites.ts:101]: But with `siteUrl` ends with a `/` and `path` starts with a `/` you will have `//` in your URL, right?

Comment (dcsaszar) [reply src/config/scrivitoSites.ts:101]: Yes, but how can you make `siteUrl` end with `/`?

**Review by apepper — APPROVED** [review]
LGTM ([e9dbe0d](https://github.com/Scrivito/scrivito-portal-app/pull/708/commits/e9dbe0d7994eb0d91eb3b8752246566785d6caeb))

**Pull request merged**